doc/services/ipc/ipc_service/backends/ipc_service_icbmsg.rst

```diff
@@ -0,0 +1,88 @@
+.. _ipc_service_backend_icbmsg:
+
+ICMsg with dynamically allocated buffers backend
+################################################
+
+This backend is built on top of the :ref:`ipc_service_backend_icmsg`.
+Data transferred over this backend travels in dynamically allocated buffers on shared memory.
+The ICMsg just sends references to the buffers.
+It also supports multiple endpoints.
+
+This architecture allows for overcoming some common problems with
+other backends (mostly related to multithread access and zero-copy).
+This backend provides an alternative with no significant limitations.
+
+Overview
+========
+
+The shared memory is divided into two parts.
+One is reserved for the ICMsg and the other contains equal-sized blocks.
+The number of blocks is configured in the devicetree.
+
+The data sending process is following:
+
+* The sender allocates one or more blocks.
+  If there are not enough sequential blocks,
+  it waits using the timeout provided in the parameter which also includes K_FOREVER and K_NO_WAIT.
+* The allocated blocks are filled with data. For the zero-copy case,
+  this is done by the caller, otherwise, it is copied automatically.
```

Comment:
```suggestion
* The allocated blocks are filled with data.
  For the zero-copy case, this is done by the caller, otherwise, it is copied automatically.
```